### DISCUSSION TITLE
Control maps should get more attention on the wiki

### DISCUSSION THREAD

**Discussion by Czebosak** [TokisanGames/Terrain3D#515] [Questions]
Control maps are very important if you want to control the terrain from scripts, so why are they at the bottom of the wiki in the development category where people don't look? The control maps are mentioned in the texturing category, but it's not enough. I think that the control map should be moved to the usage or the advanced usage category and should have their own short section in the texturing category.

**Comment by TokisanGames** [comment]
You don't know the user base. >80% are very inexperienced people. The level of difficulty from the "Advanced topics" like collision or Navigation to doing bitwise operations on a uint32_t is an order of magnitude greater. Most users won't even read the API. Moving the control map format into the advanced section won't do most users any good.

The control map is in the development category for people who are likely to also read the system architecture, shader design, and API immediately adjacent to it. Not only that, it's interwoven throughout the documentation:

https://terrain3d.readthedocs.io/en/latest/docs/controlmap_format.html
https://terrain3d.readthedocs.io/en/latest/api/class_terrain3dregion.html#class-terrain3dregion-property-control-map
https://terrain3d.readthedocs.io/en/latest/api/class_terrain3dutil.html
and 10 functions here:
https://terrain3d.readthedocs.io/en/latest/api/class_terrain3ddata.html#class-terrain3ddata-method-get-control

**Reply by Czebosak** [reply]
Okay, I'm sorry. The more I think about it the more stupid I feel. I searched on youtube, google, github issues, github discussions and the wiki, only after scrolling back and forth on the TerrainStorage class definition, only after the second time I checked the Maps enum I checked what the control map does. I remember using the web browser function to search for texture and I have no idea how I didn't see https://terrain3d.readthedocs.io/en/latest/api/class_terrain3ddata.html#class-terrain3ddata-method-get-control-base-id, I even remember looking at this method on the list and I didn't connecting the dots. I can't comprehend how I didn't see all those signs. I'm sorry. Although wouldn't it make more sense to name the method get_control_texture_id instead of get_control_base_id?

**Reply by TokisanGames** [reply]
There are two texture ids, base and overlay, and a blend value. So it could be get_control_base_texture_id(), but since the function is only syntactic sugar, long names defeat the purpose. And we don't want to get confused with get_texture_id().
